webapp/channels/src/components/admin_console/data_grid/data_grid_row.tsx

```diff
@@ -14,43 +14,41 @@ type Props = {
     row: Row;
 }
 
-class DataGridRow extends React.Component<Props> {
-    renderCell(row: Row, column: Column) {
-        const style: CSSProperties = {};
-        if (column.width) {
-            style.flexGrow = column.width;
-        }
-
-        if (column.textAlign) {
-            style.textAlign = column.textAlign;
-        }
-
-        if (column.overflow) {
-            style.overflow = column.overflow;
-        }
-
-        return (
-            <div
-                key={column.field}
-                className={classNames('DataGrid_cell', column.className)}
-                style={style}
-            >
-                {row.cells[column.field]}
-            </div>
-        );
+const renderCell = (row: Row, column: Column) => {
+    const style: CSSProperties = {};
+    if (column.width) {
+        style.flexGrow = column.width;
     }
 
-    render() {
-        const cells = this.props.columns.map((col) => this.renderCell(this.props.row, col));
-        return (
-            <div
-                className='DataGrid_row'
-                onClick={this.props.row.onClick}
-            >
-                {cells}
-            </div>
-        );
+    if (column.textAlign) {
+        style.textAlign = column.textAlign;
     }
+
+    if (column.overflow) {
+        style.overflow = column.overflow;
+    }
+
+    return (
+        <div
+            key={column.field}
+            className={classNames('DataGrid_cell', column.className)}
+            style={style}
+        >
+            {row.cells[column.field]}
+        </div>
+    );
+}
+
+const DataGridRow = ({ row, columns}: Props) => {
+    const cells = columns.map((col) => renderCell(row, col));
+    return (
+        <div
+            className='DataGrid_row'
+            onClick={row.onClick}
+        >
+            {cells}
+        </div>
+    ); 
 }
 
-export default DataGridRow;
+export default DataGridRow;
```

Comment:
It wasn't a pure component before, so it is not expected to use `React.memo` here, but @larkox what do you think? looks like the component is, in fact, pure, so we can make memoize this with `React.memo`.

Reply:
I saw that in the example PRs and did try it. I resulted in `<DataGrid>` changing to`<Memo(DataGrid)>` (since it imports `DataGridRow`). This made me think that using `React.memo` in `DataGridRow` would require `DataGrid` to also be re-written into a functional component (which is beyond the scope of this PR - happy to do it in another one though).

It also passed all the `webapp` tests.

Reply:
Since it is a "list-like" component, I think it is better to memoize at this level. A change in the list (adding one more element) can easily execute a re-render. If it is not memoized it will have to go through the `columns.map` for every element in the list, so may be troublesome.

Probably the lists this is handling are small enough and static enough for this not to be a problem (if not, I hope we would have noticed), but seems like this approach is just better. So let's go with it.

Regarding the filing tests, probably only the snapshots will change, so you should be able to update them without any issue running in the channels folder `npm run test -- -u -- path/to/test/file.test.ts`

Reply:
Memoised and modifid the snapshot.